ssc/cmod_pvsamv1.cpp

```diff
@@ -2337,8 +2338,14 @@ void cm_pvsamv1::exec()
                 }
 			}
 			// Update battery with final gen to compute grid power
-			if (en_batt)
-				batt->update_grid_power(*this, PVSystem->p_systemACPower[idx], p_load_full[idx], idx);
+            if (en_batt) {
+                if (batt->is_outage_step(idx % nrec)) {
+                    batt->update_grid_power(*this, PVSystem->p_systemACPower[idx], p_crit_load_in[idx % nrec], idx);
```

Comment:
How dows this work? If the grid is out, how is there any grid power to update?

Reply:
This is responsible for updating the interconnection loss variable, given the critical load and the updated PV generation, given the set of AC losses that weren't available when the battery is first run.